python/chemiscope/jupyter.py

```diff
@@ -2,29 +2,32 @@
 import json
 import gzip
 import ipywidgets
-from traitlets import Unicode, Bool
+from traitlets import Unicode, Bool, Dict
 
 from .input import create_input
 
 
-@ipywidgets.register
-class ChemiscopeWidget(ipywidgets.DOMWidget, ipywidgets.ValueWidget):
-    _view_name = Unicode("ChemiscopeView").tag(sync=True)
-    _view_module = Unicode("chemiscope-widget").tag(sync=True)
-
+class ChemiscopeWidgetBase(ipywidgets.DOMWidget, ipywidgets.ValueWidget):
     data = Unicode().tag(sync=True)
     has_metadata = Bool().tag(sync=True)
 
+    # synchronized settings from the JS side. You can assign to this field to
+    # change what's being displayed by chemiscope, but you need to assign a full
+    # dictionary (`widget.settings["map"]["x"]["property"] = "foo"` will not
+    # work, but `widget.settings = updated_settings` will).
+    settings = Dict().tag(sync=True)
+
     def __init__(self, data, has_metadata):
         super().__init__()
         self.data = json.dumps(data)
         self.has_metadata = has_metadata
+        self.settings = {}
 
     def save(self, path):
         """
-        Save the dataset displayed by this :py:class:`ChemiscopeWidget` as JSON
-        to the given ``path``. If ``path`` ends with ``.gz``, the file is
-        written as gzip compressed JSON string.
+        Save the dataset displayed by this widget as JSON to the given ``path``.
```

Comment:
Shall we stress that this will be the same JSON created when showing the widget, and won't be updated in terms of style?

Reply:
Also, shall we add a `save_settings` to save the instantaneous settings similar to what can be done with the webpage version?

Reply:
> Shall we stress that this will be the same JSON created when showing the widget, and won't be updated in terms of style?

This is no longer the case: https://github.com/lab-cosmo/chemiscope/pull/219/commits/a211d42301bef340218acdc13f2930f6625e7fa8#diff-71fe19e6aa531b95aac2bd2f498dc21dee83eb341831b70c40c887d8bb904d73R39-R43